File: swift/Sources/SignalClient/zkgroup/ReceiptCredentialRequest.swift
//
// Copyright 2020-2021 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

import Foundation
import SignalFfi

public class ReceiptCredentialRequest: ByteArray {
  public required init(contents: [UInt8]) throws {
    try super.init(contents, checkValid: signal_receipt_credential_request_check_valid_contents)
  }
}

File: java/java/src/main/java/org/signal/zkgroup/receipts/ReceiptCredentialRequest.java
//
// Copyright 2020-2021 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

package org.signal.zkgroup.receipts;

import org.signal.zkgroup.InvalidInputException;
import org.signal.zkgroup.internal.ByteArray;
import org.signal.client.internal.Native;

public final class ReceiptCredentialRequest extends ByteArray {

  public static final int SIZE = 97;

  public ReceiptCredentialRequest(byte[] contents) throws InvalidInputException {
    super(contents, SIZE);
    Native.ReceiptCredentialRequest_CheckValidContents(contents);
  }

  public byte[] serialize() {
    return contents.clone();
  }

}
